src/lib/swapper/swappers/ThorchainSwapper/endpoints.ts

```diff
@@ -31,37 +56,173 @@ export const thorchainApi: SwapperApi = {
-    return await getSignTxFromQuote({
-      tradeQuote,
-      receiveAddress,
-      affiliateBps,
-      chainSpecific,
-      ...fromOrXpub,
-      supportsEIP1559,
+    accountType,
+  }: GetUnsignedUtxoTransactionArgs): Promise<BTCSignTx> => {
+    const utxoChainAdapter = assertGetUtxoChainAdapter(chainId)
+
+    // TODO: pull these from db using id so we don't have type zoo and casting hell
+    const { steps, memo } = tradeQuote as ThorEvmTradeQuote
+    const { accountNumber, sellAmountIncludingProtocolFeesCryptoBaseUnit, sellAsset, feeData } =
+      steps[0]
+
+    const { vault, opReturnData } = await getUtxoThorTxInfo({
+      sellAsset,
+      xpub,
+      memo,
+    })
+
+    return utxoChainAdapter.buildSendApiTransaction({
+      value: sellAmountIncludingProtocolFeesCryptoBaseUnit,
+      xpub: xpub!,
+      to: vault,
+      accountNumber,
+      chainSpecific: {
+        accountType,
+        opReturnData,
+        // TODO: split up getTradeQuote into separate function per chain family to negate need for cast
+        satoshiPerByte: (feeData.chainSpecific as UtxoFeeData).satsPerByte,
+      },
     })
   },
 
+  getUnsignedCosmosSdkTransaction: async ({
+    tradeQuote,
+    chainId,
+    from,
+  }: GetUnsignedCosmosSdkTransactionArgs): Promise<StdSignDoc> => {
+    // TODO: pull these from db using id so we don't have type zoo and casting hell
+    const { steps, memo } = tradeQuote as ThorEvmTradeQuote
+    const { sellAmountIncludingProtocolFeesCryptoBaseUnit, sellAsset, feeData } = steps[0]
+
+    const fromThorAsset = sellAsset.chainId === KnownChainIds.ThorchainMainnet
+
+    // TODO: split up getTradeQuote into separate function per chain family to negate need for cast
+    const gas = (feeData.chainSpecific as CosmosSdkFeeData).estimatedGasCryptoBaseUnit
+    const networkFee = feeData.networkFeeCryptoBaseUnit ?? '0'
+
+    const { fee, msg, account } = await (async () => {
+      if (fromThorAsset) {
+        const fee: StdFee = {
+          amount: [{ amount: deductOutboundRuneFee(networkFee), denom: 'rune' }],
+          gas,
+        }
+
+        // https://dev.thorchain.org/thorchain-dev/concepts/memos#asset-notation
+        const msg: AminoMsg = {
+          type: 'thorchain/MsgDeposit',
+          value: {
+            coins: [{ asset: 'THOR.RUNE', amount: sellAmountIncludingProtocolFeesCryptoBaseUnit }],
+            memo,
+            signer: from,
+          },
+        }
+
+        const api = new cosmossdk.thorchain.V1Api(
+          new cosmossdk.thorchain.Configuration({
+            basePath: getConfig().REACT_APP_UNCHAINED_THORCHAIN_HTTP_URL,
+          }),
+        )
+
+        const account = await api.getAccount({ pubkey: from })
+
+        return { fee, msg, account }
+      }
+
+      const fee: StdFee = {
+        amount: [{ amount: networkFee, denom: 'uatom' }],
+        gas,
+      }
+
+      const daemonUrl = getConfig().REACT_APP_THORCHAIN_NODE_URL
```

Comment:
Do we want to make this an explicit branch on Cosmos ChainId or maybe even AssetId since it's the only THOR-supported asset from that chain? So we can then throw in case this is neither a RUNE send nor a CosmosHub Tx

Reply:
yeah I've added a switch on assetId for rune and atom